packages/atclient/src/atclient.c

```diff
@@ -270,14 +270,136 @@ exit: { return ret; }
 }
 
 int atclient_get_publickey(const atclient *atclient, const atclient_atkey *atkey, char *value, const size_t valuelen,
-                           size_t *valueolen) {
+                           size_t *valueolen, bool bypasscache) {
   int ret = 1;
 
-  // TODO: implement
+  // 1. initialize variables
+  atclient_atstr cmdbuffer;
+  atclient_atstr_init_literal(&cmdbuffer, 4096, "plookup:");
+
+  atclient_atstr atkeystr;
+  atclient_atstr_init(&atkeystr, ATKEY_GENERAL_BUFFER_SIZE);
+
+  const size_t recvlen = 4096;
+  atclient_atbytes recv;
+  atclient_atbytes_init(&recv, recvlen);
+
+  cJSON *root = NULL;
+
+  if(atkey->atkeytype != ATCLIENT_ATKEY_TYPE_PUBLICKEY) {
+    ret = 1;
+    atclient_atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "atkey->atkeytype != ATKEYTYPE_PUBLIC\n");
+    goto exit;
+  }
+
+  // 2. build plookup: command
+  if(bypasscache) {
+    ret = atclient_atstr_append(&cmdbuffer, "bypassCache:true:");
+    if (ret != 0) {
+      atclient_atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "atclient_atstr_append: %d\n", ret);
+      goto exit;
+    }
+  }
+
+  ret = atclient_atstr_append(&cmdbuffer, "all:");
+  if (ret != 0) {
+    atclient_atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "atclient_atstr_append: %d\n", ret);
+    goto exit;
+  }
+
+  ret = atclient_atkey_to_string(*atkey, atkeystr.str, atkeystr.len, &atkeystr.olen);
+  if (ret != 0) {
+    atclient_atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "atclient_atkey_to_string: %d\n", ret);
+    goto exit;
+  }
+
+  char *atkeystrwithoutpublic = NULL;
+  char *ptr = strstr(atkeystr.str, "public:");
+  if(ptr != NULL) {
+    atkeystrwithoutpublic = ptr + strlen("public:");
+  } else {
+    atclient_atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "Could not find \"public:\" from string \"%s\"\n", atkeystr.str);
+    goto exit;
+  }
+
+  ret = atclient_atstr_append(&cmdbuffer, atkeystrwithoutpublic);
+  if (ret != 0) {
+    atclient_atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "atclient_atstr_append: %d\n", ret);
+    goto exit;
+  }
+
+  ret = atclient_atstr_append(&cmdbuffer, "\r\n");
```

Comment:
With `snprintf` this could be something like:
```
char *bypasscachestr = (bypasscache) ? "bypassCache:true:" : "";
...
snprintf(cmdbuffer, cmdbufferlen, "plookup:%sall:%s\r\n", bypasscachestr, atkeystrwithoutpublic);
```

Reply:
https://github.com/atsign-foundation/at_c/commit/23737d4cb224619c3f40678a1faeeeb43d558900

Reply:
Resolving this.. thanks @realvarx